include/drivers/hwinfo.h

```diff
@@ -72,7 +72,13 @@ ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length);
  *
  * This routine retrieves the flags that indicate why the device was reset.
  *
- * Multiple calls to this routine will return the same value, unless
+ * On some platforms the reset cause flags accumulate between successive resets
+ * and this routine may return multiple flags indicating causes of all resets
+ * since the last power event. If you need to retrieve the cause for only
```

Comment:
```suggestion
 * since the last reset event. If you need to retrieve the cause for only
```

Reply:
If the hardware is sticky then it will return all reset flags since the last POR. It doesn't clear the flags between resets.

Reply:
The intention was not clear. Why not write power-on-reset event, then?

Reply:
I'll change it to "since the device was powered on".